pypose/lietensor/lietensor.py

```diff
@@ -29,100 +33,103 @@
                      'index_select', 'masked_select', 'index_copy', 'index_copy_',
                      'select', 'select_scatter', 'index_put','index_put_', 'copy_']
 
-class LieType:
+class LieType(ABC):
     '''LieTensor Type Base Class'''
     def __init__(self, dimension, embedding, manifold):
         self._dimension = torch.Size([dimension]) # Data dimension
         self._embedding = torch.Size([embedding]) # Embedding dimension
         self._manifold  = torch.Size([manifold])  # Manifold dimension
 
     @property
-    def dimension(self):
+    def dimension(self) -> torch.Size:
         return self._dimension
 
     @property
-    def embedding(self):
+    def embedding(self) -> torch.Size:
         return self._embedding
 
     @property
-    def manifold(self):
+    def manifold(self) -> torch.Size:
         return self._manifold
 
     @property
-    def on_manifold(self):
+    def on_manifold(self) -> bool:
         return self.dimension == self.manifold
 
-    def add_(self, input, other):
+    def add_(self, input: LieTensor, other: torch.Tensor) -> torch.Tensor:
```

Comment:
I believe `add_` returns `LieTensor` as [doc](https://pypose.org/docs/main/generated/pypose.add/#pypose.add). This one is used internally for updating LieTensor by gradients via various optimizers.